splunk-otel-android/src/main/java/com/splunk/rum/CurrentNetwork.java

```diff
@@ -48,18 +63,44 @@ public String toString() {
 
     @Override
     public boolean equals(Object o) {
-        if (this == o) {
-            return true;
-        }
-        if (!(o instanceof CurrentNetwork)) {
-            return false;
-        }
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
         CurrentNetwork that = (CurrentNetwork) o;
-        return state == that.state && Objects.equals(subType, that.subType);
+        return Objects.equals(carrier, that.carrier)
+                && state == that.state
+                && Objects.equals(subType, that.subType);
     }
 
     @Override
     public int hashCode() {
-        return Objects.hash(state, subType);
+        return Objects.hash(carrier, state, subType);
+    }
+
+    @SuppressWarnings("NullAway")
+    @Nullable
```

Comment:
`getSubType()` returns an `Optional` -- should we make it return a `@Nullable` value instead? So that the API in this class looks uniform

Reply:
Yeah, not sure why it was that way originally. Will make consistent and have a null party.